### PR TITLE
[Slider][ThumbTrack] Fix event handling.

### LINKED ISSUE #8758: [Slider] Clean-up internal code and fix logic related to discrete/continuous mode
The MDCSlider changes for showing tick marks in continuous mode left a bit of a mess in the code internals.

## Acceptance Criteria
0. The Slider/ThumbTrack correctly reports continuous/non-continuous events as it did before this work.
1. The Slider and ThumbTrack components clearly separate continuous/discrete from presentation/visuals (like tick marks).
2. No existing client use cases are affected.
3. Migration APIs and README are prepared to remove some of the "implicit" discrete behavior.

## Details

Specific to continuous/non-continuous event reporting, the following failures were identified with manual testing. So far I cannot determine how to perform unit tests that propagate UIControl actions.

Here are the results of tests I performed at commit `5516c5f6c9a8267020369eb5c55eaf4225b9d679` to generate the "OUTCOME" columns. I performed the same manual tests at `628b2ecb9020b0287f44b384b93946977fd07b8a` to determine the "Pass/Fail" column.

numberOfDiscrete | Continuous | shouldDisplayValueLabel | OUTCOME Events | OUTCOME Snapping | Pass/Fail
-- | -- | -- | -- | -- | --
0 | NO | YES | onUp | None | FAIL - Shows Value + Animates
2 | NO | YES | onUp | Snaps | Pass
0 | YES | YES | onMove | None | Pass
2 | YES | YES | onSnap | Snaps | Fail - Continuous + NOsnap
0 | NO | NO | onUp | None | Fail - Animates!
2 | NO | NO | onUp | Snaps | Pass
0 | YES | NO | onMove | None | Pass
2 | YES | NO | onUp | Snaps | Fail - Continuous + Nosnap

---
#### Internal data
- Associated internal bug: [b/144162364](http://b/144162364)

### PR DESCRIPTION
**Pull request by romoore**
Slider/ThumbTrack event handling was broken by the way track tick visibility
was implemented. This PR fixes that event handling.

Fixes #8758

### CODE REVIEW

File: components/Slider/src/MDCSlider.m [hunk 1]
```diff
@@ -584,7 +582,7 @@ - (void)accessibilityIncrement {
   if (self.enabled) {
     CGFloat range = self.maximumValue - self.minimumValue;
     CGFloat adjustmentAmount = kSliderAccessibilityIncrement * range;
-    if (!self.continuous && self.numberOfDiscreteValues > 1) {
+    if (self.numberOfDiscreteValues > 1) {
```
Comment (yarneo) [inline components/Slider/src/MDCSlider.m]: please add a check for if (discrete) here too

File: components/Slider/src/MDCSlider.m [hunk 2]
```diff
@@ -602,7 +600,7 @@ - (void)accessibilityDecrement {
...
```
Comment (yarneo) [inline components/Slider/src/MDCSlider.m]: here too

**Review by yarneo — APPROVED** [review]
aside from comments, LGTM

File: components/Slider/src/MDCSlider.m (hunk 2 above)
Comment (romoore) [reply components/Slider/src/MDCSlider.m]: Done. Please take a look at the Docs and implementation to see if you agree with how I handled the new configurations.

File: components/Slider/src/MDCSlider.m (hunk 1 above)
Comment (romoore) [reply components/Slider/src/MDCSlider.m]: Done.

**Pull request merged**